lib/vendors/axis-communications/sv_vendor_axis_communications.c

```diff
@@ -250,6 +274,64 @@ verify_and_parse_certificate_chain(sv_vendor_axis_communications_t *self)
   return status;
 }
 
+/* Deserializes |attestation| into |attestation_report|.
+ * The structure of |attestation| is:
+ *   - header (2 bytes)
+ *   - version (2 bytes)
+ *   - object_id (4 bytes)
+ *   - attestation_list_length (1 byte) NOTE: Struct only supports one list item
+ *   - attestation_list
+ *     - timestamp (12 bytes)
+ *     - signature_size (2 bytes)
+ *     - signature (|signature_size| bytes) */
+static svi_rc
+deserialize_attestation(sv_vendor_axis_communications_t *self)
+{
+  assert(self);
+
+  uint8_t *attestation_ptr = (uint8_t *)self->attestation;
+  size_t signature_size = 0;
+
+  if (!self->attestation) return SVI_VENDOR;
```

Comment:
Maybe we should have the check for (!self->attestation) above row 292 where it's used the first time.

Reply:
Great catch!